merino/web/api_v1.py

```diff
@@ -111,7 +119,11 @@ async def suggest(
     response = SuggestResponse(
         suggestions=suggestions,
         request_id=correlation_id.get(),
-        client_variants=client_variants.split(",") if client_variants else [],
+        # client_variant restriction: set to remove duplicates, [0:CLIENT_VARIANT_MAX] to
+        # limit potential reflection back to client.
+        client_variants=list(set(client_variants.split(",")))[:CLIENT_VARIANT_MAX]
```

Comment:
```suggestion
        client_variants=client_variants.split(",", maxsplit=CLIENT_VARIANT_MAX))[:-1]  # `[:-1]` to drop the trailing string
```
`str.split()` takes a second argument `maxsplit` to specify that at most `maxsplit` splits are done, i.e. the list will have at most `maxsplit+1` elements. Also, I don't think we need to check for duplicates here as the API spec demands that all variants should be unique. Duplicates can only be caused by either program errors or malicious requests, which should be handled by `CLIENT_VARIANT_MAX`. What do you think?

Reply:
Solid recommendation, I had considered the kwarg `maxsplit` as I just worked with it on another task. 

Since this recommended implementation would require `[:CLIENT_VARIANT_MAX]` tacked on anyways as an additional filter to the list, using it both in `maxsplit` and then on the list afterward I think decreases readability.  Also, `maxsplit` would already place a limitation on the number of elements so calling that after wouldn't have an effect on individual trailing str, just the number of elements in the list.

 I am also with you that only malicious requests or errors could result in duplicates being passed, but I opted for the option to apply the same logic to filter them as for providers. Was thinking in terms of defense in depth for this, though the risk is very very low.`CLIENT_VARIANT_MAX` will just set the minimum number of possibilities, but does not take into consideration duplicates, or am I confused? What do you think the advantages and disadvantages are to leaving the list(set()) explicit coercion followed by filtering the list length? Happy to do whatever.

Reply:
I think using `maxsplit` here could at least save a list creation if the `client_variants` contains an excessive amount of CSVs. Yeah, that `[:CLIENT_VARIANT_MAX]` is ugly, but without it, the trailing str (if any) will be included. Do you have better way to handle that?

Reply:
> CLIENT_VARIANT_MAX will just set the minimum number of possibilities, but does not take into consideration duplicates, or am I confused?

Correct, my comment completely ignored duplicates.

> What do you think the advantages and disadvantages are to leaving the list(set()) explicit coercion followed by filtering the list length?

So if don't care about duplicates, then we don't need that `set()`, also `str.split()` already returns a list so we don't need the outer `list()` either. Lastly, if we just do `str.split(",", maxsplit=CLIENT_VARIANT_MAX)`, given a string with more than CLIENT_VARIANT_MAX variants, it will return the trailing str as the last item of the list (i.e. for `1,2,3, ..., {CLIENT_VARIANT_MAX}, ..., {CLIENT_VARIANT_MAX+N}`, it returns `['1', '2', '3', ... 'CLIENT_VARIANT_MAX', 'CLIENT_VARIANT_MAX+1,...CLIENT_VARIANT_MAX+N]`). But the point of having this constant is to grab up to `CLIENT_VARIANT_MAX` of variants, so that's why I put the `[:CLIENT_VARIANT_MAX]` bit for that. Does that make sense?

Reply:
I am still not convinced that `list(set(client_variants.split(",")))[:CLIENT_VARIANT_MAX]` is the most effective way to guard against those malformed values.

How about we leave it as is for now and use another followup to handle API parameter validations as I mentioned in [here](https://github.com/mozilla-services/merino-py/pull/153/files#r1060950041)? For example, if we apply a max string length to `client_variants`, then we don't need to limit the variant number anymore. Let me know what you think.

Reply:
Ach @ncloudioj , yeah agreed, wasn't ready for full review there as I was looking at some alternatives to that which I would push. I was intending, baring any additional insights or approaches, to go with what you recommend, which looks like what I will do now as I couldn't think of something better.

Thanks for the clarification on API parameter validations, I just thought you were referring to the Dynaconf Validators for the defined `client_variants_max` which I added, which too is useful!

 I'll get this adjusted as you recommend. Thank you!